scripts/mdbook-forc-documenter/src/commands.rs

```diff
@@ -0,0 +1,108 @@
+use crate::formatter::{format_header_line, format_line};
+use anyhow::{anyhow, Result};
+use std::collections::HashMap;
+use std::ffi::OsString;
+use std::process;
+
+pub fn possible_forc_commands() -> Vec<String> {
+    let mut possible_commands = Vec::new();
+    let output = process::Command::new("forc")
+        .arg("--help")
+        .output()
+        .expect("Failed running forc --help");
+
+    let output_str = String::from_utf8_lossy(&output.stdout);
+    let lines = output_str.lines();
+
+    let mut has_parsed_subcommand_header = false;
+
+    for line in lines {
+        if has_parsed_subcommand_header {
+            let (command, _) = line.trim().split_once(' ').unwrap_or(("", ""));
+            possible_commands.push(command.to_string());
+        }
+        if line == "SUBCOMMANDS:" {
+            has_parsed_subcommand_header = true;
+        }
+    }
+
+    possible_commands
+}
+
+pub fn get_contents_from_commands(commands: &Vec<String>) -> HashMap<String, String> {
```

Comment:
```suggestion
pub fn get_contents_from_commands(commands: &[String]) -> HashMap<String, String> {
```

Whenever you have a `&Vec<T>`, 99% of the time you want a `&[T]` instead.

This is because the only distinction between a `Vec` and a slice is that a `Vec` is growable, however it cannot be grown behind an immutable reference. Arguments that take a slice are strictly more flexible, as they support not just `&Vec`s via auto-deref, but other slices, fixed-size arrays or other abstractions over slices.

This isn't super important for this PR, but just worth keeping in mind and is related to your comment [here](https://github.com/FuelLabs/sway/pull/1582#discussion_r876589051).